frontend/src/scenes/pipeline/hogfunctions/mapping/HogFunctionMapping.tsx

```diff
@@ -0,0 +1,162 @@
+import { IconPlus, IconPlusSmall, IconTrash } from '@posthog/icons'
+import { LemonButton, LemonLabel } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { Group } from 'kea-forms'
+import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+import { getDefaultEventName } from 'lib/utils/getAppContext'
+import { ActionFilter } from 'scenes/insights/filters/ActionFilter/ActionFilter'
+import { MathAvailability } from 'scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow'
+
+import { groupsModel } from '~/models/groupsModel'
+import { EntityTypes, FilterType, HogFunctionConfigurationType, HogFunctionMappingType } from '~/types'
+
+import { hogFunctionConfigurationLogic } from '../hogFunctionConfigurationLogic'
+import { HogFunctionInputs } from '../HogFunctionInputs'
+
+export function HogFunctionMapping(): JSX.Element | null {
+    const { groupsTaxonomicTypes } = useValues(groupsModel)
+    const { useMapping, showSource } = useValues(hogFunctionConfigurationLogic)
+
+    if (!useMapping) {
+        return null
+    }
+
+    return (
+        <LemonField name="mappings">
+            {({ value, onChange }) => {
+                const mappings = (value ?? []) as HogFunctionMappingType[]
+                return (
+                    <>
+                        {mappings.map((mapping, index) => (
+                            <div key={index} className="border bg-bg-light rounded p-3 space-y-2">
```

Comment:
More of a follow up comment - this new thing breaks the design pattern of having clear panels for different sections. I might fly by and bring some opinionated UI changes to see if we can make it a little cleaner. Definitely a comment on my work, not yours 😅

Reply:
Definitely feel free to step all over my "designer with 2 left hands" work (to not throw backend engineers under the bus again)